infra/modules/providers/azure/app-service/main.tf

```diff
@@ -0,0 +1,28 @@
+data "azurerm_resource_group" "appsvc" {
+  name      = "${var.service_plan_resource_group_name}"
+}
+
+data "azurerm_app_service_plan" "appsvc" {
+  name                    = "${var.service_plan_name}"
+  resource_group_name     = "${data.azurerm_resource_group.appsvc.name}"
+}
+
+resource "azurerm_app_service" "appsvc" {
+  name                = "${var.app_service_name[count.index]}"
+  resource_group_name = "${data.azurerm_resource_group.appsvc.name}"
+  location            = "${data.azurerm_resource_group.appsvc.location}"
+  app_service_plan_id = "${data.azurerm_app_service_plan.appsvc.id}"
+  tags                = "${var.resource_tags}"
+  count               = "${length(var.app_service_name)}"
+
+  app_settings {
+    DOCKER_REGISTRY_SERVER_URL      = "${var.docker_registry_server_url[count.index]}"
```

Comment:
This feels rigid as in most cases developers will be using the same docker registry. In the instance of deploying multiple app services, one would have to redefine the docker credentials each time.

Reply:
It will be rigid, if we don't give developers the option. That is the reason each app-service has these settings.

Reply:
my point is that if you're using the same registry across 5 app services then someone is going to have to define the docker registry 5 times(all containing the same values)... Also, this seems like an unnecessary step for images hosted publicly on docker hub.

Reply:
Yes but a developer now has the option of adding it for an app service. If not they are optional and can be kept empty. Presuming that the docker registry is going to be same is against best practices which kind of takes away the option of flexibility for the developer.

Reply:
not sure why this was marked as resolved as this approach would force me to provide my docker credentials for public images hosted on docker hub.